src/test/java/test/ReadinessLivenessProbesTest.java

```diff
@@ -0,0 +1,102 @@
+package test;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import test.helm.Helm;
+import test.model.Product;
+import java.util.Map;
+import static test.jackson.JsonNodeAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ReadinessLivenessProbesTest {
+
+    private Helm helm;
+
+    @BeforeEach
+    void initHelm(TestInfo testInfo) {
+        helm = new Helm(testInfo);
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Product.class, names = {"bamboo_agent"}, mode = EnumSource.Mode.EXCLUDE)
+    void test_readiness_probe_disabled(Product product) throws Exception {
+        final var resources = helm.captureKubeResourcesFromHelmChart(product, Map.of(
+                product + ".readinessProbe.enabled", "false"));
+
+        assertThat(resources.getStatefulSet(
+                product.getHelmReleaseName()).getContainer().get("readinessProbe")).isEmpty();
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Product.class, names = {"bamboo_agent"}, mode = EnumSource.Mode.EXCLUDE)
+    void test_liveness_probe_disabled(Product product) throws Exception {
+        final var resources = helm.captureKubeResourcesFromHelmChart(product, Map.of(
+                product + ".livenessProbe.enabled", "false"));
+
+        assertThat(resources.getStatefulSet(
+                product.getHelmReleaseName()).getContainer().get("livenessProbe")).isEmpty();
+        assertThat(resources.getStatefulSet(
+                product.getHelmReleaseName()).getContainer().get("startupProbe")).isEmpty();
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Product.class, names = {"bamboo_agent"}, mode = EnumSource.Mode.EXCLUDE)
+    void test_liveness_probe_defaults(Product product) throws Exception {
+        final var resources = helm.captureKubeResourcesFromHelmChart(product, Map.of());
+
+        assertEquals("5", resources.getStatefulSet(
+                product.getHelmReleaseName()).getContainer().get("livenessProbe").get("periodSeconds").asText());
+        assertEquals("1", resources.getStatefulSet(
+                product.getHelmReleaseName()).getContainer().get("livenessProbe").get("failureThreshold").asText());
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Product.class, names = {"bamboo_agent"}, mode = EnumSource.Mode.EXCLUDE)
+    void test_liveness_probe_overrides(Product product) throws Exception {
+        final var resources = helm.captureKubeResourcesFromHelmChart(product, Map.of(
+                product + ".livenessProbe.periodSeconds", "10",
```

Comment:
I would probably use some more random number (let's say we change to `10` in future, then this test won't really test the overrides).

Reply:
Fixed